### PR TITLE
config: fix write config change to config file

### PR DESCRIPTION
**Pull request by NingLin-P**
### What problem does this PR solve?

Problem Summary:

This pr fix:
- "Invalid cross-device link" error when trying to rewrite config file
- wrong toml format cause by rewrite config to empty file

### Check List 

Tests 

- Manual test 

### Release note 
- Fix "Invalid cross-device link" error when trying to rewrite config file
- Fix wrong toml format cause by rewrite config to empty file

### CODE REVIEW

**Comment by NingLin-P** [comment]
> Could you add some tests to cover the fix?

Added, PTAL.

**Comment by NingLin-P** [comment]
PTAL @overvenus @rleungx

File: src/config.rs [hunk 1]
```diff
@@ -2356,15 +2365,17 @@ fn to_config_change(change: HashMap<String, String>) -> CfgResult<ConfigChange>
...
-                f = f.replace("-", "_");
-            }
+                field.replace("-", "_")
+            };
             return match typed.get(&f) {
-                None => Err(format!("unexpect fields: {}", f).into()),
-                Some(ConfigValue::Skip) => Err(format!("config {:?} can not be change", f).into()),
+                None => Err(format!("unexpect fields: {}", field).into()),
+                Some(ConfigValue::Skip) => {
+                    Err(format!("config {} can not be change", field).into())
```
Comment (rleungx) [inline src/config.rs:2377]:
```suggestion
                    Err(format!("config {} can not be changed", field).into())
```

File: src/config.rs [hunk 2]
```diff
@@ -2334,8 +2334,17 @@ pub fn persist_config(config: &TiKvConfig) -> Result<(), String> {
     Ok(())
 }
 
-pub fn write_config<P: AsRef<Path>>(tem_dir: P, path: P, content: &[u8]) -> CfgResult<()> {
-    let tmp_cfg_path = tem_dir.as_ref().join(TMP_CONFIG_FILE);
+pub fn write_config<P: AsRef<Path>>(path: P, content: &[u8]) -> CfgResult<()> {
```
Comment (rleungx) [inline src/config.rs:2337]: Is it possible that the config file in the root path, like `/config_file`?

Comment (NingLin-P) [reply src/config.rs:2337]: Yes, it is possible.

**Pull request merged**